R/acs_dm.R

```diff
@@ -156,10 +172,49 @@ dm.demand <- function(B17024,
   assertthat::assert_that(all(df$working_pov_rate <= 100, na.rm = TRUE))
   assertthat::assert_that(all(df$n_kids_lt5_working_under200_pct <= df$n_kids_lt6_working_under200_pct, na.rm = TRUE))
   assertthat::assert_that(all(df$n_kids_lt4_working_under200_pct <= df$n_kids_lt6_working_under200_pct, na.rm = TRUE))
-  
+
   return(df)
 }
 
+#' @title Add default attributes for ACS tables
+#' @inheritParams child_care_db
+acs_attr <- function(acs_year,
+                     acs_state_code,
+                     acs_geography,
+                     acs_county,
+                     lt6 = "002",
+                     pth) {
+
+  cls <- function(table, geography, lt6, other_lt6, pth) {
+    structure(
+      list(table = table,
+           year = acs_year,
+           state = acs_state_code,
+           geography = acs_geography,
+           county = acs_county,
+           lt6 = paste(table, lt6, sep = "_"),
+           other_lt6 = paste(table, other_lt6, sep = "_"),
+           pth = pth),
+      class = c(table, geography))
```

Comment:
This is where the classes get assigned. We assign each object a class of geography, so if the geography changes then the tables are automatically reassigned. Then each class has its own function, so only that function gets called.